src/components/Modal.js

```diff
@@ -0,0 +1,121 @@
+import React from "react";
+import BusRouteDetails from "./BusRouteDetails";
+
+export default function Modal({ selectedRoute, closeModal }) {
+
+    //data percentiles
+
+  // i averaged the data for these percentiles in excel. you can use this key by using the index as the first digit of the percentile block, so percentileKeys[2] will be percentiles from 20-29% 
+  // FIX ME : eventually we're going to want these numbers rendered dynamically when the backend/data updates
+
+
+  var percentileKeys = [62, 72, 75, 77, 80, 83, 86, 90, 93, 95]
+
+
+
+  // take all the ratios from the data and average out each line's mean bus reliablility
+
+  // most recent data only shows relability ("acc") for weekday bus lines, so setting totalAcc to the lines ratio. Leaving below code in because this is a short term change and i dont want to re-write it!!
+
+  // const weekdayRoutes = selectedRoute.filter(
+  //   (route) => route.properties.day_type === "wk"
+  // );
+  // const sundayRoutes = selectedRoute.filter(
+  //   (route) => route.properties.day_type === "sun"
+  // );
+  // const saturdayRoutes = selectedRoute.filter(
+  //   (route) => route.properties.day_type === "sat"
+  // );
+
+  // const weekdayAcc =
+  //   ((weekdayRoutes[0].properties.ratio + weekdayRoutes[1].properties.ratio) /
+  //     2) *
+  //   100;
+  // const saturdayAcc =
+  //   saturdayRoutes.length > 0
+  //     ? ((saturdayRoutes[0].properties.ratio +
+  //         saturdayRoutes[1].properties.ratio) /
+  //         2) *
+  //       100
+  //     : 0;
+  // const sundayAcc =
+  //   sundayRoutes.length > 0
+  //     ? ((sundayRoutes[0].properties.ratio + sundayRoutes[1].properties.ratio) /
+  //         2) *
+  //       100
+  //     : 0;
+
+  // const totalDataPoints =
+  //   (saturdayAcc > 0 ? 1 : 0) +
+  //   (sundayAcc > 0 ? 1 : 0) +
+  //   (weekdayAcc > 0 ? 1 : 0);
+  const totalAcc = (selectedRoute[0].properties.ratio)*100;
+
+  //round the percentages to the nearest ten
+
+  function round10(x) {
+    let digits = x.toString().split("");
+    if (digits[1] >= 5) {
+      digits = [Number(digits[0]) + 1, 0];
+    } else {
+      digits[1] = 0;
+    }
+    digits = digits.map((x) => Number(x));
+    console.log(parseInt(digits.join("")))
+    return parseInt(digits.join(""));
+  }
+
+  //reduce fraction to lowest common denonminatior
+
+  function reduce(numerator, denominator) {
```

Comment:
I'd love to put unit testing around the `reduce` and `calcBusFraction`, but that could be a suggestion for a future improvement.